tests/test_cmd_unify.py

```diff
@@ -188,6 +251,126 @@ def test_unify(self):
+
+        identities = jsmith.identities
+        identities.sort(key=lambda x: x.id)
+
+        self.assertEqual(len(identities), 7)
+
+        id_ = identities[1]
+        self.assertEqual(id_.email, 'JSmith@example.com')
+        self.assertEqual(id_.source, 'mls')
+
+        id_ = identities[3]
+        self.assertEqual(id_.email, 'jsmith@example.com')
+        self.assertEqual(id_.source, 'scm')
+
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, UNIFY_DEFAULT_OUTPUT)
+        self.assertFalse(os.path.exists(self.custom_log))
+
+    @patch('os.path.expanduser')
+    def test_unify_from_default_log(self, mock_expanduser):
+        """Test unify method when reading matches from the default log file"""
+
+        original_log = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data/matches.log')
+        shutil.copyfile(original_log, self.custom_log)
+        mock_expanduser.return_value = self.custom_log
+
+        before = api.unique_identities(self.db)
+        self.assertEqual(len(before), 6)
+
+        self.assertTrue(os.path.exists(self.custom_log))
+        code = self.cmd.unify(matching='default')
+        self.assertEqual(code, CMD_SUCCESS)
+
+        after = api.unique_identities(self.db)
+        self.assertEqual(len(after), 5)
+
+        # jsmith identities with same email address
+        jsmith = after[0]
+        self.assertEqual(jsmith.uuid, '178315df7941fc76a6ffb06fd5b00f6932ad9c41')
+
+        identities = jsmith.identities
+        identities.sort(key=lambda x: x.id)
+
+        self.assertEqual(len(identities), 7)
+
+        id_ = identities[1]
+        self.assertEqual(id_.email, 'JSmith@example.com')
+        self.assertEqual(id_.source, 'mls')
+
+        id_ = identities[3]
+        self.assertEqual(id_.email, 'jsmith@example.com')
+        self.assertEqual(id_.source, 'scm')
+
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, UNIFY_DEFAULT_OUTPUT)
+        self.assertFalse(os.path.exists(self.custom_log))
+
+    @patch('sortinghat.api.merge_unique_identities')
+    def test_unify_save_to_custom_log(self, mock_merge_unique_identities):
+        """Test unify method when saving matches to a custom log"""
+
+        mock_merge_unique_identities.side_effect = Exception
+
+        self.assertFalse(os.path.exists(self.custom_log))
+        with self.assertRaises(Exception):
+            self.cmd.unify(matching='default', log_path=self.custom_log)
+        self.assertTrue(os.path.exists(self.custom_log))
+
+        with open(self.custom_log, 'r') as f:
+            count_objs = 0
+            for line in f.readlines():
+                matches_obj = json.loads(line.strip("\n"))
+                self.assertTrue(all([isinstance(m, six.string_types) for m in matches_obj['identities']]))
+                self.assertFalse(matches_obj['processed'])
+                count_objs += 1
+
+                self.assertTrue(count_objs, 5)
```

Comment:
There's a mistake here. This assertion should be outside of the loop. Actually, it's hiding an error. The method to use is `assertEqual` and not `assertTrue`. That's why I don't like to use `assertTrue`.